plugins/sharding/logical_count.rb

```diff
@@ -2,42 +2,27 @@ module Groonga
   module Sharding
     class LogicalCountCommand < Command
       register("logical_count",
-               [
-                 "logical_table",
-                 "shard_key",
-                 "min",
-                 "min_border",
-                 "max",
-                 "max_border",
-                 "filter",
-                 "post_filter",
-               ])
+              [
+                "logical_table",
+                "shard_key",
+                "min",
+                "min_border",
+                "max",
+                "max_border",
+                "filter",
+                "post_filter",
+              ])
 
       def run_body(input)
-        enumerator = LogicalEnumerator.new("logical_count", input)
-        counter = Counter.new(input, enumerator.target_range)
-        total = 0
-        have_shard = false
-
+        context = ShardCountContext.new(input)
         begin
-          enumerator.each do |shard, shard_range|
-            have_shard = true
-            counter.count_pre(shard, shard_range)
-          end
-          total += counter.count
-          unless have_shard
-            message =
-              "[logical_count] no shard exists: " +
-              "logical_table: <#{enumerator.logical_table}>: " +
-              "shard_key: <#{enumerator.shard_key_name}>"
-            raise InvalidArgument, message
-          end
+          counter = Counter.new(context)
+          count_result = counter.count
```

Comment:
Why did you rename this from `total`?
It seems that `count_result` isn't better than `total`.

Reply:
I have fixed it.